pyanaconda/modules/payloads/payload/dnf/dnf_manager.py

```diff
@@ -892,6 +895,44 @@ def set_repository_enabled(self, repo_id, enabled):
             repo.disable()
             log.info("The '%s' repository is disabled.", repo_id)
 
+    def read_system_repositories(self):
```

Comment:
Similar issue as with the next comment. I see the repositories disabling as side effect of reading the repositories. The method name clearly states that it reads the repository but it doesn't tell you anything about disabling them.

I guess the best way to solve this is to split these to `read_system_repositories` and `disable_system_repositories`?

Reply:
The system repositories should be always available and disabled by default, unless they are required by the installation source or enabled by a user. Introducing additional states for the system repositories can cause problems.